examples/with-grafana/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Trendyol/kafka-konsumer"
+	"strconv"
+	"sync/atomic"
+	"time"
+)
+
+type user struct {
+	ID   int
+	Name string
+}
+
+var messages = []user{
+	{ID: 1, Name: "foo"},
+	{ID: 2, Name: "bar"},
+	{ID: 3, Name: "baz"},
+	{ID: 4, Name: "qux"},
+	{ID: 5, Name: "fred"},
+}
+
+func main() {
+	// retryMap stores the number of retries for each message
+	var retryMap = make(map[int]int, len(messages))
+	for _, message := range messages {
+		retryMap[message.ID] = 0
+	}
+
+	// create new kafka producer
+	producer, _ := kafka.NewProducer(kafka.ProducerConfig{
+		Writer: kafka.WriterConfig{
+			Brokers: []string{"localhost:29092"},
+		},
+	})
+
+	// produce messages at 1 seconds interval
+	ticker := time.NewTicker(1 * time.Second)
+	quit := make(chan struct{})
+	go func() {
+		var i uint64
+		for {
+			select {
+			case <-ticker.C:
+				message := messages[atomic.LoadUint64(&i)]
+				bytes, _ := json.Marshal(message)
+
+				_ = producer.Produce(context.Background(), kafka.Message{
+					Topic: "konsumer",
+					Key:   []byte(strconv.Itoa(message.ID)),
+					Value: bytes,
+				})
+
+				if message.ID == messages[len(messages)-1].ID {
+					quit <- struct{}{}
+					return
+				}
+
+				atomic.AddUint64(&i, 1)
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	consumerCfg := &kafka.ConsumerConfig{
+		APIEnabled:  true,
+		Concurrency: 1,
+		Reader: kafka.ReaderConfig{
+			Brokers: []string{"localhost:29092"},
+			Topic:   "konsumer",
+			GroupID: "konsumer.group.test",
+		},
+		RetryEnabled: true,
+		RetryConfiguration: kafka.RetryConfiguration{
+			Brokers:       []string{"localhost:29092"},
+			Topic:         "konsumer-retry",
+			StartTimeCron: "*/1 * * * *",
+			WorkDuration:  50 * time.Second,
+			MaxRetry:      3,
+		},
+		ConsumeFn: func(message kafka.Message) error {
```

Comment:
Shall we extract this function here for simplicity

Reply:
I think extracting this section makes the code more complex